plugins/inventory/lxd.py

```diff
@@ -0,0 +1,1066 @@
+    version_added: "2.5.0"
+    author: "Frank Dornheim (@conloos)"
+    options:
+        plugin:
+            description: Token that ensures this is a source file for the 'lxd' plugin.
+            required: true
+            choices: [ 'community.general.lxd' ]
+        url:
+            description:
+            - The unix domain socket path or the https URL for the lxd server.
+            - Sockets in filesystem have to start with C('unix:').
+            - Mostly C(unix:/var/lib/lxd/unix.socket) or C(unix:/var/snap/lxd/common/lxd/unix.socket).
+            default: unix:/var/snap/lxd/common/lxd/unix.socket
+            required: false
+            type: str
+        client_key:
+            description:
+            - The client certificate key file path.
+            required: false
+            aliases: [ key_file ]
+            default: $HOME/.config/lxc/client.key
+            type: path
+        client_cert:
+            description:
+            - The client certificate file path.
+            required: false
+            aliases: [ cert_file ]
+            default: $HOME/.config/lxc/client.crt
+            type: path
+        trust_password:
+            description:
+            - The client trusted password.
+            - You need to set this password on the lxd server before
+                running this module using the following command
+                C('lxc config set core.trust_password <some random password>')
+                See U(https://www.stgraber.org/2016/04/18/lxd-api-direct-interaction/).
+            - If I(trust_password) is set, this module send a request for authentication before sending any requests.
+            required: false
+            type: str
+        state:
+            description: Filter the container according to the current status.
+            required: false
+            type: str
+            default: none
+            choices: [ 'STOPPED', 'STARTING', 'RUNNING', 'none' ]
+        prefered_container_network_interface:
+            description:
+            - If a container has multiple network interfaces, select which one is the prefered as pattern.
+            - Combined with the first number that can be found e.g. 'eth' + 0.
+            required: false
+            type: str
+            default: eth
+        prefered_container_network_family:
+            description:
+            - If a container has multiple network interfaces, which one is the prefered by family.
+            - Specify C(inet) for IPv4 and C(inet6) for IPv6.
+            required: false
+            type: str
+            default: inet
+            choices: [ 'inet', 'inet6' ]
+        groups:
+            description:
+            - Create Groups by the following keywords location, pattern, network_range, os, release, profile, vlanid.
+            - See example for syntax.
+            required: false
+            type: json
+            default: none
+        dumpdata:
```

Comment:
I'm not sure such an option should exist. Why would users need this?

Reply:
I have not all variants of lxd setups. So for example i have no cluster setup. If a user have an issue, i want to give him a way to create a raw data set. But i can drop that option and implement the generation of that data by -vvvv?!
If you don't like the idea i can drop this support but it makes more difficult to help users.

Reply:
You should better add an `_info` module which allows to retrieve that data, instead of adding this option.

Reply:
Do you have an example of this implementation ?

Reply:
You can look at https://github.com/ansible-collections/community.general/blob/main/plugins/modules/net_tools/pritunl/pritunl_org_info.py as an example. It is a small module which basically allows you to query some (or all, if none is specified) objects from the API.